src/Illuminate/Queue/QueueManager.php

```diff
@@ -295,6 +295,88 @@ public function setApplication($app)
         return $this;
     }
 
+    /**
+     * Pause a queue by name.
+     *
+     * @param  string  $queue
+     * @param  int  $ttl
+     * @return void
+     */
+    public function pause($queue, $ttl = 86400)
+    {
+        $cache = $this->app['cache']->store();
+
+        $cache->put("queue_paused:{$queue}", true, $ttl);
```

Comment:
Operation is not atomic here. In some concurrent situation (pausing two queues same time) threir may be such case:

* first process getPausedQueues returns []
* second proccess getPausedQueues returns []
* first process storePausedQueuesList sets ['queue1']
* second process storePausedQueuesList sets ['queue2']

finaly queue_paused_list has ['queue2'] and not ['queue1', 'queue2'] as expected.

One way is to use some sort of locks or isolation here.
But maybe it would be better for siplicicty not to provide PauseList functionality at all. Pausing and resuming certain queue (and single isolated cache key) would be enough. Currently framwork does not have any  functions to get all existing queues list, so listing just paused queues is redundant.